pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/assignment/segment/OfflineSegmentAssignment.java

```diff
@@ -52,61 +61,53 @@ protected int getReplication(TableConfig tableConfig) {
   @Override
   public List<String> assignSegment(String segmentName, Map<String, Map<String, String>> currentAssignment,
       Map<InstancePartitionsType, InstancePartitions> instancePartitionsMap) {
+    SegmentAssignmentStrategy segmentAssignmentStrategy = _assignmentStrategyMap.get(InstancePartitionsType.OFFLINE);
+    // Need to check assignmentStrategy for Dim Tables as following pre conditions state would return false
+    if (segmentAssignmentStrategy instanceof DimTableSegmentAssignmentStrategy) {
```

Comment:
I still don't get why we need to specially handle `DimTable`. You mean `Preconditions.checkState(instancePartitions !=null)` will throw the exception?

The existing code doesn't go through the precondition but I think that it's ok to go through. In the actual code path, we always provide both `currentAssignment , instancePartitionsMap`. If some of the tests fails complaining that the instancePartitions is null, we should fix forward. 

In reality, we should not face this issue because having null instancePartitionMap means that there's no available server for this table.

Reply:
discussed offline